project_task_stage_allow_timesheet/tests/test_project_task_stage_allow_timesheet.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestProjectTaskStageAllowTimesheet(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestProjectTaskStageAllowTimesheet, cls).setUpClass()
+        cls.AnalyticLine = cls.env["account.analytic.line"]
+
+        cls.employee_1 = cls.env.ref("hr.employee_qdp")
+        cls.project_1 = cls.env.ref("project.project_project_1")
+        cls.task_1 = cls.env.ref("project.project_1_task_1")
+
+        cls.stage_new = cls.env.ref("project.project_stage_0")
+
+    def test_01_stage_allow_timesheet(self):
+        self.task_1.stage_id = self.stage_new
+
+        values = {
+            "name": "test",
+            "project_id": self.project_1.id,
+            "task_id": self.task_1.id,
+            "employee_id": self.employee_1.id,
+        }
+        self.AnalyticLine.create(values)
+
+        self.stage_new.allow_timesheet = False
+        with self.assertRaises(ValidationError) as e, self.env.cr.savepoint():
+            self.AnalyticLine.create(values)
+
+        self.assertIn(
+            "You can't link a timesheet line to a task if its stage doesn't allow it.",
+            e.exception.name,
```

Comment:
2023-06-13 11:16:10,337 241 WARNING odoo py.warnings: /opt/odoo/odoo/exceptions.py:33: DeprecationWarning: UserError attribute 'name' is a deprecated alias to args[0]
  File "/opt/odoo-venv/bin/coverage", line 8, in <module>
    sys.exit(main())
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 974, in main
    status = CoverageScript().command_line(argv)
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 684, in command_line
    return self.do_run(options, args)
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 8[61](https://github.com/OCA/timesheet/actions/runs/5254729608/jobs/9493705294?pr=588#step:8:62), in do_run
    runner.run()
  File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/execfile.py", line 211, in run
    exec(code, main_mod.__dict__)
  File "/opt/odoo-venv/bin/odoo", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/opt/odoo/setup/odoo", line 8, in <module>
    odoo.cli.main()
  File "/opt/odoo/odoo/cli/command.py", line 66, in main
    o.run(args)
  File "/opt/odoo/odoo/cli/server.py", line 180, in run
    main(args)
  File "/opt/odoo/odoo/cli/server.py", line 173, in main
    rc = odoo.service.server.start(preload=preload, stop=stop)
  File "/opt/odoo/odoo/service/server.py", line 1399, in start
    rc = server.run(preload, stop)
  File "/opt/odoo/odoo/service/server.py", line 579, in run
    rc = preload_registries(preload)
  File "/opt/odoo/odoo/service/server.py", line 1299, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "<decorator-gen-14>", line 2, in new
  File "/opt/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/opt/odoo/odoo/modules/registry.py", line 90, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/opt/odoo/odoo/modules/loading.py", line 488, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/opt/odoo/odoo/modules/loading.py", line 372, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/opt/odoo/odoo/modules/loading.py", line 290, in load_module_graph
    test_results = loader.run_suite(suite, module_name)
  File "/opt/odoo/odoo/tests/loader.py", line 80, in run_suite
    suite(results)
  File "/usr/lib/python3.10/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/opt/odoo/odoo/tests/suite.py", line 44, in run
    test(result)
  File "/usr/lib/python3.10/unittest/case.py", line [65](https://github.com/OCA/timesheet/actions/runs/5254729608/jobs/9493705294?pr=588#step:8:66)0, in __call__
    return self.run(*args, **kwds)
  File "/opt/odoo/odoo/tests/common.py", line 2[69](https://github.com/OCA/timesheet/actions/runs/5254729608/jobs/9493705294?pr=588#step:8:70), in run
    super().run(result)
  File "/opt/odoo/odoo/tests/case.py", line 216, in run
    self._callTestMethod(testMethod)
  File "/opt/odoo/odoo/tests/case.py", line 1[84](https://github.com/OCA/timesheet/actions/runs/5254729608/jobs/9493705294?pr=588#step:8:85), in _callTestMethod
    method()
  File "/__w/timesheet/timesheet/project_task_stage_allow_timesheet/tests/test_project_task_stage_allow_timesheet.py", line 38, in test_01_stage_allow_timesheet
    e.exception.name,
  File "/opt/odoo/odoo/exceptions.py", line 33, in name
    warnings.warn(

Reply:
Thanks for the comments @CRogos